packages/ra-core/src/controller/input/useReferenceArrayInputController.ts

```diff
@@ -115,7 +114,7 @@ const useReferenceArrayInputController = ({
     // the component displaying the currently selected records may fail
     const finalMatchingReferences =
         matchingReferences && matchingReferences.length > 0
-            ? matchingReferences.concat(finalReferenceRecords)
+            ? mergeReferences(matchingReferences, finalReferenceRecords)
```

Comment:
This was creating warnings about duplicate keys: selected values that were also in the list of possible choices were displayed twice!